celery/concurrency/gevent.py

```diff
@@ -17,15 +19,17 @@
 
 def apply_timeout(target, args=(), kwargs={}, callback=None,
                   accept_callback=None, pid=None, timeout=None,
-                  timeout_callback=None, Timeout=Timeout,
+                  soft_timeout=None, timeout_callback=None, Timeout=Timeout,
                   apply_target=base.apply_target, **rest):
+    soft = None
     try:
-        with Timeout(timeout):
-            return apply_target(target, args, kwargs, callback,
-                                accept_callback, pid,
-                                propagate=(Timeout,), **rest)
-    except Timeout:
-        return timeout_callback(False, timeout)
+        with Timeout(soft_timeout, SoftTimeLimitExceeded) as soft:
```

Comment:
Can you please explain something here? Since `soft_timeout` is lower than `timeout`:

```python
with gevent.Timeout(5, SoftTimeLimitExceeded):
      with gevent.Timeout(6):
            gevent.sleep(5.2)
```

The above code block won't always raise `SoftTimeLimitExceeded`?